docs/_docs/reference/contextual/more-givens.md

```diff
@@ -0,0 +1,204 @@
+---
+layout: doc-page
+title: "Other Forms Of Givens"
+nightlyOf: https://docs.scala-lang.org/scala3/reference/contextual/givens.html
+---
+
+The concept of given instances is quite general. This page covers forms of givens that were not treated before.
+
+## Simple Structural Givens
+
+Some givens simply instantiate a class without needing an alias or additional member declarations. Example:
+
+```scala
+class IntOrd extends Ord[Int]:
+  def compare(x: Int, y: Int) =
+    if x < y then -1 else if x > y then +1 else 0
+
+given IntOrd()
+```
+In this case, the given clause consists of just a class creation expression, such as `IntOrd()` above.
+
+## Conditional Givens with Parameters
+
+Conditional givens can also be defined with parameters. Example:
+```scala
+given (config: Config) => Factory = MemoizingFactory(config)
+```
+Here, `(config: Config)` describes a context parameter expressing a condition: We can synthesize a given `Factory` _provided_ we can synthesize a given `config` of type `Config`.
+
+Type parameters and context parameters can be combined. For instance the `listOrd` instance above could alternatively be expressed like this:
+```scala
+given listOrd: [T] => Ord[T] => Ord[List[T]]:
+  ...
+  def compare(x: List[T], y: List[T]) = ...
+```
+As the example shows, each parameter section is followed by an `=>`.
+
+It is also possible to name context parameters:
+```scala
+given listOrd: [T] => (ord: Ord[T]) => Ord[List[T]]:
+  ...
+```
+
+## By Name Givens
+
+We sometimes find it necessary that a given alias is re-evaluated each time it is called. For instance, say we have a mutable variable `curCtx` and we want to define a given that returns the current value of that variable. A normal given alias will not do since by default given aliases are mapped to lazy vals.
+
+In general, we want to avoid re-evaluation of givens. But there are situations like the one above where we want to specify _by-name_ evaluation instead. This is achieved by writing a conditional given with an empty parameter list:
```

Comment:
```suggestion
Though in general we want to avoid re-evaluating a given, there are situations where such a re-evaluation may be necessary. For instance, say we have a mutable variable `curCtx` and we want to define a given that returns the current value of that variable. A normal given alias will not do since by default given aliases are mapped to lazy vals. In this case, we can specify a _by-name_ evaluation insteadby writing a conditional given with an empty parameter list:
```